enumerables.rb

```diff
@@ -0,0 +1,134 @@
+module Enumerable
+  # My Each Method
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self.class == Array ? self : to_a
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter])
+      counter += 1
+    end
+    arr
+  end
+
+  # My Each With Index
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self.class == Array ? self : to_a
+    counter = 0
+    while counter < arr.length
+      yield(arr[counter], counter)
+      counter += 1
+    end
+    arr
+  end
+
+  # My Select
+  def my_select(arr)
+    new_array = []
+    arr.my_each do |index_value|
+      new_array << index_value if yield(index_value)
+    end
+    new_array
+  end
```

Comment:
## Enumerator error
- [ ] It should raise a "Enumerator" error when no block or argument is given

## Range and array self
- [ ] Test with hashes and ranges, then compare with original `select` Ruby method
- [ ] Fix your method to work with the tests. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)

**Example:**

`test_file.rb`
```ruby
(0..9).my_select{
# or
Range.new(0, 9)


```

## Optional argument or none argument
- [ ] Add optional argument or remove to this method. Follow this [sample](https://gist.github.com/bertil291utn/5a7e6a73351c38ceb9781a9a37b58cf6)


```ruby
def my_select?(argument = nil)
def my_select?
```